sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsRelationshipAsyncTest.java

```diff
@@ -268,39 +269,77 @@ public void relationshipListOperationWithMultiplePages(HttpClient httpClient, Di
-                    .doOnSuccess(s -> createdRelationshipIds.add(relationshipId))
-                    .doOnTerminate(createRelationshipsLatch::countDown)
-                    .subscribe();
+                StepVerifier.create(
+                    asyncClient.createRelationship(
+                        floorTwinId,
+                        relationshipId,
+                        deserializeJsonString(floorContainsRoomPayload, BasicRelationship.class),
+                        BasicRelationship.class)).verifyComplete();
+                createdRelationshipIds.add(relationshipId);
             }
 
-            createRelationshipsLatch.await(MAX_WAIT_TIME_ASYNC_OPERATIONS_IN_SECONDS, TimeUnit.SECONDS);
+            // Create multiple incoming relationships to the floor. Typically a room would have relationships to multiple
+            // different floors, but for the sake of test simplicity, we'll just add multiple relationships from the same room
+            // to the same floor.
+            for (int i = 0; i < pageSize + 1; i++) {
+                String relationshipId = ROOM_CONTAINED_IN_FLOOR_RELATIONSHIP_ID + this.testResourceNamer.randomUuid();
+                StepVerifier.create(
+                    asyncClient.createRelationship(
+                        roomTwinId,
+                        relationshipId,
+                        deserializeJsonString(roomContainedInFloorPayload, BasicRelationship.class),
+                        BasicRelationship.class)).verifyComplete();
+                createdRelationshipIds.add(relationshipId);
+            }
 
-            AtomicInteger pageCount = new AtomicInteger();
+            AtomicInteger outgoingRelationshipsPageCount = new AtomicInteger();
             // List models in multiple pages and verify more than one page was retrieved.
             StepVerifier.create(asyncClient.listRelationships(floorTwinId, BasicRelationship.class).byPage())
                 .thenConsumeWhile(
                     page -> {
-                        pageCount.getAndIncrement();
-                        logger.info("content for this page " + pageCount);
+                        outgoingRelationshipsPageCount.getAndIncrement();
+                        logger.info("content for this page " + outgoingRelationshipsPageCount);
                         for (BasicRelationship relationship : page.getValue()) {
                             logger.info(relationship.getId());
                         }
+
+                        if (page.getContinuationToken() != null) {
+                            assertEquals(RELATIONSHIP_PAGE_SIZE_DEFAULT, page.getValue().size(), "Unexpected page size for a non-terminal page");
+                        }
+
                         return true;
                     })
                 .verifyComplete();
 
-            int finalPageCount = pageCount.get();
+            assertTrue(outgoingRelationshipsPageCount.get() > 1, "Number of pages must be more than one.");
```

Comment:
Can this be something more like:

```java
    assertThat("Number of pages must be more than 1",
           outgoingRelationshipsPagecount.get(),
           greaterThan(1));
```

In .NET at least, doing a comparison rather than asserting true or false gets you better test failure output.